streaming/src/java/org/apache/hive/streaming/StreamingConnection.java

```diff
@@ -60,6 +64,16 @@
    */
   void commitTransaction() throws StreamingException;
 
+  /**
+   * Commit a transaction to make the writes visible for readers. Include
+   * other partitions that may have been added independently.
+   *
+   * @param partitions - extra partitions to commit.
+   * @throws StreamingException - if there are errors when committing the open transaction.
+   */
+  void commitTransactionWithPartition(@Nullable Set<String> partitions)
```

Comment:
Are there plans to backport this patch to branch-3? if so we have to consider backward compatibility as many public interfaces/packages/classes have changed. Nifi for example is using many of the public interfaces which will break with these changes. Maybe we can add a new interface that extends StreamingConnection or provide default method for the new interfaces.